### PR TITLE
Add ability to specify different project and key storage types

### PR DESCRIPTION
**Pull request by jyaworski**
http://rundeck.org/docs/administration/project-setup.html

### CODE REVIEW

**Comment by jyaworski** [comment]
Tag @liamjbennett I'd really like to merge this, but I also want code review. How does it look, minus tests?

File: templates/rundeck-config.erb [hunk 1]
```diff
@@ -32,8 +32,13 @@ grails.mail.default.from = "<%= @mail_config['defaults.from'] %>"
 <%- end -%>
 grails.serverURL = "<%= @grails_server_url %>"
 rundeck.clusterMode.enabled = "<%= @clustermode_enabled %>"
-<%- if @file_keystorage_dir %>
-rundeck.storage.provider."1".config.baseDir = "<%= @file_keystorage_dir %>"
-rundeck.storage.provider."1".path = "/"
+
+rundeck.projectsStorageType = "<%= @projects_storage_type %>"
```
Comment (liamjbennett) [inline templates/rundeck-config.erb]: Don't you also need to add other project storage config here .e.g. `rundeck.config.storage.provider."1"` ?

**Comment by liamjbennett** [comment]
Ref: http://rundeck.org/docs/administration/storage-facility.html

File: templates/rundeck-config.erb (hunk 1 above)
Comment (jyaworski) [inline templates/rundeck-config.erb]: I tested this and that's all I needed.

**Comment by jyaworski** [comment]
Note: this works, but the projects define doesn't work with it. That will need docs, or someone to write queries around it. @liamjbennett ideas?

**Comment by jyaworski** [comment]
My final thought is that we instead wrap the cli commands to do the project define... I'll work on that next in a new PR.

**Pull request merged**